NOTICE.txt

```diff
@@ -0,0 +1,3980 @@
+
+
+
+Overview
+--------
+
+
+Spyder incorporates code and image assets from various external sources,
+each of which is covered by its own license and copyright notice.
+These files are listed here with their corresponding authors,
+permission notices, disclaimers, and copyright statements; these are also
+included in the files' directory and/or their headers, as appropriate.
+
+Any file not listed here is covered by Spyder's MIT (Expat) license,
+described in the LICENSE.txt file in this repository and Spyder's __init__.py.
+
+Images, icons and font gyphs that include trademarked elements are used for
+informational purposes only, and do not necessarily constitute any endorsement,
+association or ownership by or of the marks or their respective owners.
+
+All trademarks are the property of their respective owners.
+
+
+-------------------------------------------------------------------------------
+
+
+
+Instructions for Adding a New Entry
+-----------------------------------
+
+
+All files or groups of files, including source code, images, icons, and other
+assets, that originate from projects outside of the Spyder organization
+(regardless of the license), must be documented here.
+Please see the Contributing Guide, CONTRIBUTING.md, for full details on the
+criteria for inclusion and other steps you need to take to ensure transparency,
+legal clarity and license compliance.
+Always check with the Spyder team (on Github, Gitter or the Google Group)
+before adding content from an external project, and only do so when necessary.
+
+Add one entry per code repo, icon set or font face, or other cohesive
+unit within which the source material is organized and distributed as.
+New entires for icons should be in order of the number of icons used;
+other entries should be in order of inclusion.
+
+Pay careful attention to the number of blank lines before and after headings,
+horizontal rules, sections, and other elements.
+This helps keep this document semi-machine-readable and helps avoid conflicts.
+Break this document at 79 characters, and at periods (sentences) unless the
+text is copied directly from an external file (license, header, etc).
+Follow the project standards here and for added files (see CONTRIBUTING.md).
+
+Please check the list below to ensure your project's license is on it.
+If not, you'll need to add it (but double-check it with us first).
+Add its name here, and copy the plain text (broken at > 80 characters) in the
+final section at the end of this document, at the end of the list.
+
+
+Source code licenses:
+
+* MIT (Expat) License
+* 3-Clause ("Modified") BSD License
+* Python Software Foundation License Version 2
+* Apache License 2.0
+
+
+Additional asset licenses:
+
+* GNU LGPL 2.1 (with GPL 2.0)
+* GNU LGPL 3.0 (with GPL 3.0)
+* Creative Commons Attribution 3.0
+* SIL Open Font License 1.1
+
+
+-------------------------------------------------------------------------------
+
+
+
+New Entry Template
+------------------
+
+
```

Comment:
Well, you get the idea. There a are lot of unneeded blanks in this file. Please remove them to make it more tight.

Reply:
I slimmed down the lines at the top (as in your first few comments), as well as trimming 1-2 blanks (down to the minimal 2 required to maintain consistency and clarity between major elements of each entry) between the top- and section-level headings and the text below, and likewise did the same between the text and the section separator lines. However, as I elaborated in my rationale for it being plain text and not Markdown/rst above, with regard to the essentially all the remaining

> unneeded blanks

that I haven't removed  *are* needed and have meaning and purpose. For this file more than any other, meaning, purpose, and unambiguous clarity take absolute precedence over aesthetics, unless the aesthetics both serve and do not conflict with those ends.

To that end, the line breaks serve a the primary purpose of clearly establishing the textual hierarchy and unambiguously defining the format of each entry and section: Four lines between sections; three lines between entries; two lines between top-level elements of each entry. Particularly, the latter (on which the higher levels were based) was necessary due to the vital need to separate the descriptive text copied from the source and that added, which often contains blank lines as part of its verbatim native formatting (any more I slimmed down to just a single one). 

In light of your request, I carefully considered whether the two line breaks between the headings and the content, and between the content and the ``-----`` separator, could be removed, but ultimately concluded that it would be both inconsistent with the above separation and hierarchy convention; looks odd, as every other element is separated by two lines; and could also lead to some ambiguity particularly in the license full text section.

Furthermore, they are intended to aid in avoiding merge conflicts for two different PRs adding new entries by ensuring at least one blank line is present in the original for any addition of a new entry/section (one line on each side, and one between on which the change is nominally made) to avoid triggering git's conflict detection (which relies on at least one blank line between any parallel changes); I haven't extensively tested how much this helps in practice, but it certainly won't hurt. Finally, there is the practical aspect, that any further changes (which would necessarily be to the above convention, and thus have cascading effects) would involve changing many hundreds of more lines of the source, and would require touching every single entry (greatly complicating any future change that involved reverting or otherwise altering most of the 40+ commits in this PR).